lib/notifier.js

```diff
@@ -94,20 +94,39 @@ function buildBaseData(extra) {
 
 
 function buildErrorData(baseData, err, callback) {
-  parser.parseException(err, function (e, errData) {
-    if (e) {
-      return callback(e);
-    }
-    baseData.body.trace = {
-      frames: errData.frames,
-      exception: {
-        class: errData['class'],
-        message: errData.message
-      }
-    };
-
-    callback(null);
-  });
+  var errors = [];
+    
+    do {
+        errors.push(err);
+    } while((err = err.innerError) !== undefined)
```

Comment:
Since you mentioned the return callback below I also noticed that I probably want to `return cb(e);` as well. If parsing of the first exceptions fails it should halt but continue if the parsing fails for depth 2 and below (>= 2)